### PR TITLE
plonkish prelude

### PR DESCRIPTION
**Pull request by martyall**
## TLDR
There are some kimchi  / general crypto modules that have dependencies on `pickles_types`. This PR fixes this and makes them available for other crypto use cases.

## The problem
We want to share some code that is currently "owned" by `kimchi`/`pickles` with some upcoming projects, specifically `o1vm` and the compatibility layer `vinegar`. The type definitions/code is useful for plonkish proving systems, so we want to separate it in a way where it can be shared.

## The Solution
The library `pickles_types` contains many such modules, so we want to break them out into two categories:
  - general use (e.g. `nat.ml`, `vector.ml`, etc)
  - `kimchi` specific (e.g. `plonk_types.ml`)

We move the general use modules to a new library `plonkish_prelude` which has no `kimchi` dependencies. For the `kimchi` specific, we move them to `kimchi_backend_common` because this was the lowest level library that was using them directly. 

NOTE: for now we are re-exporting everything through `pickles_types` in order to not have to change tons of lines of code which depend on the `pickles_types` namespace. We can get rid of this `(1)` now, `(2)` in a future pr or `(3)` never. Ideally `(2)`

### CODE REVIEW

**Comment by martyall** [comment]
!ci-nightly-me

**Comment by martyall** [comment]
https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/3016

**Comment by dannywillems** [comment]
LGTM so far! Like it.

**Comment by dannywillems** [comment]
!ci-build-me

**Review by dannywillems — APPROVED** [review]
I love this idea. LGTM.
